### PR TITLE
feat(firestore-bigquery-export): big query partitioned tables

### PR DESCRIPTION
**Pull request by russellwheatley**
Fixes https://github.com/firebase/extensions/issues/316

1. Tested creating fresh time partitioned BigQuery tables without a problem
2. Tested "updating" the extension whilst pointing to existing non-partitioned BigQuery tables. 

The feature is backwards compatible. The BigQuery tracker package always checks if the table exists before setting the `timePartitioning` metadata for the raw changelog table and the latest view table. This also means the import script will not be affected as long as the table has already been created by the extension (which is the recommended way).

In future, we may also wish to update the `gen-schema-view` script to also allow time partitioning, but I suggest we wait for this to land and raise a separate PR.

### CODE REVIEW

**Review by dackers86 — APPROVED** [review]

File: firestore-bigquery-export/extension.yaml [hunk 1]
```diff
@@ -213,3 +213,25 @@ params:
...
+    options:
+      - label: hour
+        value: HOUR
+      - label: day
+        value: DAY
+      - label: month
+        value: MONTH 
+      - label: year
+        value: YEAR
+      - label: no table partitioning
```
Comment (Salakar) [inline firestore-bigquery-export/extension.yaml:234]:
```suggestion
      - label: none
```

File: firestore-bigquery-export/extension.yaml [hunk 2]
```diff
@@ -213,3 +213,25 @@ params:
...
+        value: false
+    default: false
```
Comment (Salakar) [inline firestore-bigquery-export/extension.yaml:236]:
```suggestion
        value: NONE
    default: NONE
```

**Review by Salakar — CHANGES_REQUESTED** [review]
LGTM with minor options naming changes

**Review by Salakar — APPROVED** [review]

File: firestore-bigquery-export/extension.yaml [hunk 3]
```diff
@@ -213,3 +213,25 @@ params:
...
+      - label: day
+        value: DAY
+      - label: month
+        value: MONTH 
+      - label: year
+        value: YEAR
+      - label: none
+        value: NONE
+    default: NONE
+    required: true
```
Comment (jhuleatt) [inline firestore-bigquery-export/extension.yaml:237]: @kevinthecheung can you please give this new param a review?

Comment (jhuleatt) [reply firestore-bigquery-export/extension.yaml:237]: Kevin is OOO. This looks good to me.

**Review by jhuleatt — APPROVED** [review]

**Pull request merged**